include/aws/common/linked_list.h

```diff
@@ -54,31 +53,70 @@ extern "C" {
     /**
      * Initializes a circular doubly-linked list.
      */
-    AWS_COMMON_API void aws_linked_list_init(struct aws_linked_list_node *head);
+    static inline AWS_COMMON_API void aws_linked_list_init(struct aws_linked_list_node *head);
 
     /**
      * Removes all the elements from the list.
```

Comment:
How useful of a function is this? How do you find all the elements that were cleared so you can free them later?

Reply:
It would be useful only in the static list case, or if the contents of the list are ints/pointers.

Reply:
The list has no contents; it is _contained within_ another list. So you still need to have something find and free list nodes. I don't think this is particularly useful given that use case.

Reply:
no big deal will remove it.